src/test/rustdoc-ui/no-run-flag.stdout

```diff
@@ -0,0 +1,12 @@
+
+running 7 tests
+test $DIR/no-run-flag.rs - f (line 11) ... ok
+test $DIR/no-run-flag.rs - f (line 14) ... ignored
+test $DIR/no-run-flag.rs - f (line 17) ... ok
+test $DIR/no-run-flag.rs - f (line 23) ... ok
+test $DIR/no-run-flag.rs - f (line 28) ... ok
+test $DIR/no-run-flag.rs - f (line 32) ... ok
+test $DIR/no-run-flag.rs - f (line 8) ... ok
```

Comment:
If it didn't run why is it okay?

Reply:
so adding the `--no-run` flag change the behavior of the test like it was annotated as \`\`\`no_run. Maybe this is unwanted and something else should be displayed.

Reply:
Ping @CraftSpider @jyn514 should we have a different text? "ok" may be confusing. Also note that we may also want a different color or the same color as "ignored".

Reply:
Hm. I don't think 'ignored' is quite right, but 'ok' is also a bit confusing. If we can make it something new, maybe something like 'built' or 'checked'? If not, I think 'ok' is more accurate than 'ignored'.
Color-wise, I'm fine with it being ok colored, as it is a requested change to behavior, not implicit, so it's requested to treat it as success. If that makes sense.

Reply:
After investigations I think that in order to do so, I have to modify the lib test. This will also affect all tests and not just doctests. Do you think this is a worthwhile investment and should this be in a different PR ? Anyway I would be interested by implementing that but the scope is a bit bigger than I anticipated.

Reply:
more precisely in prints are done in `library\test\src\console.rs:101 :  write_log_result` I will need to add a `TestResult` and change `TestDesc` and probably more.

Reply:
No, I don't think that should be part of this PR.

Reply:
What happens if both `should_test` and `no_run` is active? So it only compiles them but isn't that the same as `no_run` by itself? And it still prints "ok"? Maybe we should check again if we have any combination of options that might make this weird but right now I only see `should_test` being conflicting.

Reply:
`no_run`  by itself is useless. 
To summarize:
- if you run `rustdoc -Z unstable-options --no-run` it just build the documentation, it has the same behavior  as just `rustdoc`
- `rustdoc -Z unstable-options --no-run --test` run the doctest set. It behave as if every test had the no_run meaning they won't run. Meaning that test like 
\`\`\`
panic!()
\`\`\`
will print as ok like
\`\`\`no_run
panic!()
\`\`\`
without the `--no-run` flag

I agree that is confusing. Maybe a error should be raised if no_run is active and not `should_test`. However I think that if you run `rustdoc -Z unstable-options --no-run` your intention was probably to run the test instead of documenting your project.

I assume that in the original issue the final goal would be to be able to generate bin (with `--persist-doctests`) but without running them. 

Or maybe a another possible case would be to unify the behavior of `cargo test --no-run` (provided that down the line cargo is also modified). For instance `cargo test --all` test both regular test and doctest but `cargo test --all --no-run` skips doctest  `cargo test --doc --no-run` give an error.

Now if the explicit goal is to harmonize with regular test the output should just be empty.
Another option would for me to change the test lib such that no_run test display something like `test foo ... checked`, provided that such change is desirable (to discuss thought the RFC process ?).
Moreover test with `compile_fail` should also print something different maybe checked is also desirable.

I am new contributing for rust so I am not quite sure of what to do to solve this issue.

Reply:
> I agree that is confusing. Maybe a error should be raised if no_run is active and not should_test. However I think that if you run rustdoc -Z unstable-options --no-run your intention was probably to run the test instead of documenting your project.

I think we should give a hard error here; if people meant to run the test they can add `--test`. Could you suggest adding `--test` in the error message?

See https://github.com/rust-lang/rust/blob/202659a43d2bf9444e3d14f5917bf9cc90f72b16/src/librustdoc/config.rs#L587-L593 for an example of how to give an error if the options are inconsistent.

I think all your other changes can go in a follow-up PR, I don't think we should be changing libtest here.

Reply:
Ok thank you. I have not change libtest yet I will work on that too :)

Reply:
Please do not change libtest in this pr. I am not a maintainer and don't know who to ask to approve; finding out will delay landing this.

Reply:
Sorry I was not clear. I will do another PR for libtest